include/rs485/read_write_buffer.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <stddef.h>
+
+struct ReadWriteBuffer {
```

Comment:
It seems to me that `Buffer` is a misnomer, since this is not necessarily a memory buffer but instead something that is expected to do some sort of IO. In that case, I'd call this something like `BusIO`.